opentelemetry-sdk/tests/trace/propagation/test_jaeger_format.py

```diff
@@ -0,0 +1,120 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import opentelemetry.sdk.trace as trace
+import opentelemetry.sdk.trace.propagation.jaeger_format as jaeger_format
+import opentelemetry.trace as trace_api
+from opentelemetry import baggage
+from opentelemetry.trace.propagation.textmap import DictGetter
+
+FORMAT = jaeger_format.JaegerFormat()
+
+
+carrier_getter = DictGetter()
+
+
+def get_context_new_carrier(old_carrier, carrier_baggage=None):
+
+    ctx = FORMAT.extract(carrier_getter, old_carrier)
+    if carrier_baggage:
+        for key, value in carrier_baggage.items():
+            ctx = baggage.set_baggage(key, value, ctx)
+    parent_span_context = trace_api.get_current_span(ctx).get_span_context()
+
+    parent = trace._Span("parent", parent_span_context)
+    child = trace._Span(
+        "child",
+        trace_api.SpanContext(
+            parent_span_context.trace_id,
+            trace_api.RandomIdsGenerator().generate_span_id(),
+            is_remote=False,
+            trace_flags=parent_span_context.trace_flags,
+            trace_state=parent_span_context.trace_state,
+        ),
+        parent=parent.get_span_context(),
+    )
+
+    new_carrier = {}
+    ctx = trace_api.set_span_in_context(child, ctx)
+
+    FORMAT.inject(dict.__setitem__, new_carrier, context=ctx)
+
+    return ctx, new_carrier
+
+
+class TestJaegerFormat(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        ids_generator = trace_api.RandomIdsGenerator()
+        cls.trace_id = ids_generator.generate_trace_id()
+        cls.span_id = ids_generator.generate_span_id()
+        cls.parent_span_id = ids_generator.generate_span_id()
+        cls.serialized_uber_trace_id = jaeger_format.format_uber_trace_id(
+            cls.trace_id, cls.span_id, cls.parent_span_id, 1
+        )
+
+    def test_extract_valid_span(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        ctx = FORMAT.extract(carrier_getter, old_carrier)
+        span_context = trace_api.get_current_span(ctx).get_span_context()
+        self.assertEqual(span_context.trace_id, self.trace_id)
+        self.assertEqual(span_context.span_id, self.span_id)
+
+    def test_trace_id(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        _, new_carrier = get_context_new_carrier(old_carrier)
+        self.assertEqual(
+            self.serialized_uber_trace_id.split(":")[0],
+            new_carrier[FORMAT.TRACE_ID_KEY].split(":")[0],
+        )
+
+    def test_parent_span_id(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        _, new_carrier = get_context_new_carrier(old_carrier)
+        span_id = self.serialized_uber_trace_id.split(":")[1]
+        parent_span_id = new_carrier[FORMAT.TRACE_ID_KEY].split(":")[2]
+        self.assertEqual(span_id, parent_span_id)
+
+    def test_sampled_flag(self):
```

Comment:
I'd suggest adding tests for the other flags (debug, firehose).

As mentioned above, I believe this will fail to parse sample flags that use the alpha hex characters.

Reply:
@toumorokoshi Could you elaborate on this, please? How would we test these flags when they cannot be stored as part of the trace?

Reply:
I was thinking add a test that includes the jaeger firehose flag, which is 0x8 therefore will go beyond the single byte limit, and catch the lack of a hex conversion.

But it wouldn't have caught this because it's still within the 4 byte limit: https://github.com/jaegertracing/jaeger-client-go/pull/419/files#diff-552f47512a00afe5fc6850cc9ddc830a6daeca162750e50aab4ed549685e0253R27.

Regardless it might be good to add flags supporting the other flags (firehose).